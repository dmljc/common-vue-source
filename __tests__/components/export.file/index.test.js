/**
 * @jest-environment jsdom
 */
import { createLocalVue, mount } from '@vue/test-utils';
import Compose from '@/main';
import ExportFile from '@/components/export.file';

let customMount;
beforeAll(() => {
    const localVue = createLocalVue();
    localVue.use(Compose);
    customMount = options => mount(ExportFile, Object.assign({localVue}, options));
});

describe('export-file default props & data', () => {
    it('export-file default props & data', () => {
        const wrapper = customMount();
        window.setTimeout(() => {
            expect(wrapper.props('type')).toBe('confirm');
            expect(wrapper.vm.range).toBe(1);
            expect(wrapper.element).toMatchSnapshot();
        }, 10);
    });
});

describe('export-file customs props', () => {
    it('export-file has correct custom props behavior', () => {
        const wrapper = customMount({
            propsData: {
                title: '报表导出',
                type: 'rangeCheck',
                visible: true
            }
        });
        window.setTimeout(() => {
            expect(wrapper.element).toMatchSnapshot();
        }, 10);
    });
});

describe('export-file methods', () => {
    it('method[export] has correct behavior', (done) => {
        const wrapper = customMount({
            propsData: {
                visible: true
            },
            listeners: {
                export: (e) => {
                    expect(e.range).toBe(1);
                },
                close: () => {
                    done();
                }
            }
        });
        const buttons = wrapper.findAll('button').filter(b => b.text() === '确认导出');
        buttons.at(0).trigger('click');
        expect(wrapper.element).toMatchSnapshot();
    });
});
