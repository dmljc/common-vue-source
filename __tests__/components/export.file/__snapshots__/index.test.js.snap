// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-file methods method[export] has correct behavior 1`] = `
<div
  class="el-dialog__wrapper export-file-block"
  name="dialog-fade"
  style="z-index: 2003;"
>
  <div
    class="el-dialog"
    style="width: 680px; margin-top: 15vh;"
  >
    <div
      class="el-dialog__header"
    >
      <span
        class="el-dialog__title"
      >
        导出
      </span>
      <button
        aria-label="Close"
        class="el-dialog__headerbtn"
        type="button"
      >
        <i
          class="el-dialog__close el-icon el-icon-close"
        />
      </button>
    </div>
    <div
      class="el-dialog__body"
    >
      <div>
        <p
          class="export-confirm__item"
        >
          您确认要导出当前查询条件下的所有数据吗？
        </p>
         
        <p
          class="export-confirm__item"
        >
          确认导出后，数据将以邮件的形式发送到您的工作邮箱
        </p>
      </div>
       
      <!---->
       
    </div>
    <div
      class="el-dialog__footer"
    >
      <div
        class="f-tac"
      >
        <button
          class="el-button el-button--default"
          type="button"
        >
          <!---->
          <!---->
          <span>
            取消
          </span>
        </button>
         
        <button
          class="el-button el-button--primary"
          type="button"
        >
          <!---->
          <!---->
          <span>
            确认导出
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
